src/main/java/bio/terra/controller/ResourcesApiController.java

```diff
@@ -0,0 +1,85 @@
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.WebDataBinder;
+import org.springframework.web.bind.annotation.InitBinder;
+import org.springframework.web.bind.annotation.RequestBody;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.validation.Valid;
+import java.util.Optional;
+import java.util.UUID;
+
+@Controller
+public class ResourcesApiController implements ResourcesApi {
+
+    private final ObjectMapper objectMapper;
+    private final HttpServletRequest request;
+    private final ProfileService profileService;
+    private final BillingProfileRequestValidator billingProfileRequestValidator;
+
+    @Autowired
+    public ResourcesApiController(
+            ObjectMapper objectMapper,
+            HttpServletRequest request,
+            ProfileService profileService,
+            BillingProfileRequestValidator billingProfileRequestValidator) {
+        this.objectMapper = objectMapper;
+        this.request = request;
+        this.profileService = profileService;
+        this.billingProfileRequestValidator = billingProfileRequestValidator;
+    }
+
+    @Override
+    public Optional<ObjectMapper> getObjectMapper() {
+        return Optional.ofNullable(objectMapper);
+    }
+
+    @Override
+    public Optional<HttpServletRequest> getRequest() {
+        return Optional.ofNullable(request);
+    }
+
+    @InitBinder
+    protected void initBinder(final WebDataBinder binder) {
+        binder.addValidators(billingProfileRequestValidator);
+    }
+
+    @Override
+    public ResponseEntity<BillingProfileModel> createProfile(
+        @RequestBody BillingProfileRequestModel billingProfileRequest) {
+        return new ResponseEntity<>(profileService.createProfile(billingProfileRequest), HttpStatus.CREATED);
+    }
+
+    @Override
+    public ResponseEntity<DeleteResponseModel> deleteProfile(String id) {
+        UUID profileId = UUID.fromString(id);
+        return new ResponseEntity<>(profileService.deleteProfileById(profileId), HttpStatus.OK);
+    }
+
+    @Override
+    public ResponseEntity<EnumerateBillingProfileModel> enumerateProfiles(
+            @Valid Integer offset,
+            @Valid Integer limit) {
+        ControllerUtils.validateEnumerateParams(offset, limit);
+        EnumerateBillingProfileModel ebpm = profileService.enumerateProfiles(offset, limit);
+        return new ResponseEntity<>(ebpm, HttpStatus.OK);
+    }
+
+    @Override
+    public ResponseEntity<BillingProfileModel> retrieveProfile(String id) {
+        UUID profileId = UUID.fromString(id);
+        BillingProfileModel profileModel = profileService.getProfileById(profileId);
+        return new ResponseEntity<>(profileModel, HttpStatus.OK);
+    }
```

Comment:
As I'm reading this code, i'm wondering about authorization. if we are just using these calls internally maybe we don't need authz. but if we split this out as a separate service we definitely will. do we want to think about this later? (before it is a service on it's own?)

Reply:
Yep, I don't think we need it now. As a follow-up to this I'd like to make a profile resource in SAM with a policy that controls which users can create studies for that policy -- this would be when we could add the authz piece here.